API Arena : Complete IFlag implementation

Sulexa opened the pull request:
Hello,
I'm using your project for Screeps Arena and i found that IFlag is missing the "my" property.
https://arena.screeps.com/docs/#capture_the_flag::Flag.my
Which is a problem because in the capture the flag mode you can have yours/ennemies/neutrals flags.
So this PR add support to the My property on IFlag.
Thanks for all your work on this project, it's wonderful.

Sulexa committed:
Implement IFlag.My

thomasfn commented:
Thanks for the PR! Just a quick question - can the flag ownership ever change in any of the arenas, e.g. if you capture them? If so, the property should use the per-tick cache rather than the lifetime cache - otherwise the `my` property will never change once cached.

Sulexa commented:
Oh yeah, you're right, not in the basic mode but in the advanced mode there is some neutrals flags that could be taken back and forth by the two players. 
I will update that.

Sulexa committed:
Uses per-tick caching for flag's `My` property

Changes the caching strategy for the `My` property of flags to use per-tick caching.

This ensures the `My` property accurately reflects the flag's ownership status within the current game tick, preventing stale data.

Sulexa commented:
Corrected and tested with the tutorial 7

thomasfn commented:
There's one more thing I think we need (sorry - it's been a while since I've looked at this project and having to remember how everything works!). There's a `ClearNativeCache` virtual method on `NativeGameObject` that should be overridden (don't forget to call base) and should null out the new `My` property - otherwise it will never get cleared between ticks. e.g. implementation of creep that's clearing out the cache for other tick-variant data:  https://github.com/thomasfn/ScreepsDotNet/blob/9d7857c9d2df9486a730bc30c2b7b752a483d982/ScreepsDotNet.API/Native/Arena/NativeCreep.cs#L128

Sulexa commented:
No worry, again thank you for your great project and your help on this PR.
I will look into it

Sulexa committed:
Override CleanNativeCache and clear myCache
